micrometer-core/build.gradle

```diff
@@ -103,6 +113,9 @@ dependencies {
     testImplementation 'org.testcontainers:mongodb'
 }
 
+
+sourceSets.main.java.srcDirs += 'src/main/kotlin'
```

Comment:
I have a Java class under kotlin folder and without this it's not compiled

Reply:
I think it would work if the Java source were in `src/main/java/io/micrometer/core/instrument/kotlin` or if it stays where it is but is rewritten in Kotlin.